armsrc/Standalone/hf_bog.c

```diff
@@ -0,0 +1,278 @@
+//-----------------------------------------------------------------------------
+// This code is licensed to you under the terms of the GNU GPL, version 2 or,
+// at your option, any later version. See the LICENSE.txt file for the text of
+// the license.
+//-----------------------------------------------------------------------------
+// main code for standalone HF Sniff (and ULC/NTAG/ULEV1 pwd storing)
+//-----------------------------------------------------------------------------
+#include "hf_bog.h"
+
+#define DELAY_READER_AIR2ARM_AS_SNIFFER (2 + 3 + 8) 
+#define DELAY_TAG_AIR2ARM_AS_SNIFFER (3 + 14 + 8)
+
+// Maximum number of auth attempts per standalone session
+#define MAX_PWDS_PER_SESSION 20
+
+
+uint8_t FindOffsetInFlash() {
+	uint8_t mem[4] = { 0x00, 0x00, 0x00, 0x00 };
+	uint8_t eom[4] = { 0xFF, 0xFF, 0xFF, 0xFF };
+	uint8_t memcnt = 0;
+	
+	while (1)
+	{
+		Flash_ReadData(memcnt, mem, 4);
+		if (memcmp(mem, eom, 4) == 0) {
+			break;
+		}
+		memcnt += 4;
+	}
+	
+	return memcnt;
+}
+
+void EraseMemory()
+{
+    if (!FlashInit()){
+        return;
+    }
+    
+    Flash_CheckBusy(BUSY_TIMEOUT);
+    Flash_WriteEnable();
+    Flash_Erase4k(0,0);
+
+    if (MF_DBGLEVEL > 1) Dbprintf("[!] Erased flash!");
+    FlashStop();
+	SpinDelay(100);
+}
+
+void RAMFUNC SniffAndStore(uint8_t param) {
+	
+	// Array to store the authpwds
+	uint8_t *capturedPwds = BigBuf_malloc(4 * MAX_PWDS_PER_SESSION);
+	
+	SpinDelay(500);
+
+	/* This is actually copied from SniffIso14443a */
+	
+	iso14443a_setup(FPGA_HF_ISO14443A_SNIFFER);
+	
+	// Allocate memory from BigBuf for some buffers
+	// free all previous allocations first
+	BigBuf_free(); BigBuf_Clear_ext(false);
```

Comment:
will this free not cancel your previous malloc?

Reply:
Actually I think that this just zeroes out the Bigbuf, it doesn't leave dangling pointers, but I moved it after the free call just to be safe.

Reply:
also resets the pointer,  meaning next call to malloc will point to same area...